### PR TITLE
[API] renamed `ax` parameter for consistency

### LINKED ISSUE #4471: inconsistent parameter name for axes
while working on this I noticed we have 2 functions in the entire code base that use
`ax` instead of `axes` in their function definition.

- plot_design_matrix
- plot_contrast_matrix

```
$ grep -nr " ax=" nilearn/**/*.py

nilearn/plotting/matrix_plotting.py:339:    contrast_def, design_matrix, colorbar=False, ax=None, output_file=None
nilearn/plotting/matrix_plotting.py:459:def plot_design_matrix(design_matrix, rescale=True, ax=None, output_file=None):
```

I would suggest make those consistent with the rest of the codebase.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/4467#discussion_r1655318278_

### PR DESCRIPTION
**Pull request by muddi900**
- Closes #4471

Changes proposed in this pull request:

- Renames the all parameters names `ax` to axes.
- Changes the parameters wherever the functions are called.

### CODE REVIEW

**Comment by 7AXEL** [comment]
Check out this project: https://github.com/7AXEL/Web2APK

**Review by bthirion — COMMENTED** [review]
Thx for doing this.
But since this in an API change, we need to go through a deprecation cycle.

**Comment by Remi-Gau** [comment]
can you undo a few of those changes as they relate to the matplotlip API and not ours and they break the examples?

See for example
https://github.com/nilearn/nilearn/actions/runs/9698378496/job/26764918624?pr=4476#step:13:2060

File: examples/02_decoding/plot_miyawaki_encoding.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def index_to_xy_coord(x, y, z=10):
         )
     )
     plt.axis("off")
-    plt.colorbar(ax_im, ax=ax)
+    plt.colorbar(ax_im, axes=ax)
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_miyawaki_encoding.py:273]:
```suggestion
    plt.colorbar(ax_im, ax=ax)
```

File: examples/02_decoding/plot_miyawaki_encoding.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ def index_to_xy_coord(x, y, z=10):
     )
 )
 plt.axis("off")
-plt.colorbar(ax_im, ax=ax)
+plt.colorbar(ax_im, axes=ax)
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_miyawaki_encoding.py:303]:
```suggestion
plt.colorbar(ax_im, ax=ax)
```

**Comment by Remi-Gau** [comment]
> But since this in an API change, we need to go through a deprecation cycle.

@muddi900 

Nilearn has a decorator you can use to help with this: it will rename parameters passed by users and throw a deprecation warning.

Probably worth mentioning the change in the doc strings of the public functions as well.

See this recent PR.
https://github.com/nilearn/nilearn/pull/4470/files#r1658227494

**Comment by Remi-Gau** [comment]
2 other things to do:
- add yourself as a contributor in the citation.cff file
- update the changelog so that change to the API will be mentioned in the release notes

See the relevant section of the contributing guide: https://nilearn.github.io/stable/development.html#changelog

Let me know if something is not clear.

**Comment by muddi900** [comment]
Should I add the changes as deprecation or code?

**Comment by Remi-Gau** [comment]
> Should I add the changes as deprecation or code?

let's go with deprecation

**Comment by muddi900** [comment]
Would this be a good deprecation quote?

- :bdg-danger:`Deprecation` The parameter ``ax`` has been replaced by ``axes`` in ``plot_contrast_matrix`` and ``plot_design_matrix``. (:gh:`4476` by `Mudassir Chapra`_)

**Comment by Remi-Gau** [comment]
almost: the release should tell people when the change will become effective, this way people have time to adapt their code.

also you can use rst syntax to link directly the name of the function to wherever it is documented.


```rst
- :bdg-danger:`Deprecation` The parameter ``ax`` will be replaced by ``axes`` in :func:`nilearn.plotting.plot_contrast_matrix` and :func:`nilearn.plotting.plot_design_matrix` in release 0.13.0. (:gh:`4476` by `Mudassir Chapra`_)
```

**Comment by Remi-Gau** [comment]
> > But since this in an API change, we need to go through a deprecation cycle.
> 
> Nilearn has a decorator you can use to help with this: it will rename parameters passed by users and throw a deprecation warning.
> 
> Probably worth mentioning the change in the doc strings of the public functions as well.
> 
> See this recent PR. https://github.com/nilearn/nilearn/pull/4470/files#r1658227494

@muddi900 do you think you can add those deprecation warnings as well?

if you are not sure how to go about it, let us know and I can take care of one function to show you how it's done and then I can let you take care of the rest?

**Comment by muddi900** [comment]
What should I add as the target version?

**Comment by Remi-Gau** [comment]
> What should I add as the target version?

0.13.0

the next minor release should be 0.11.0, and we have deprecation warning for 2 minor releases to give users time to adapt.

**Review by Remi-Gau — APPROVED** [review]
Good with me

**Comment by Remi-Gau** [comment]
the CI failure is unrelated

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**